packages/orama/src/components/defaults.ts

```diff
@@ -25,17 +25,35 @@ export async function getDocumentIndexId(doc: Document): Promise<string> {
 
 export async function validateSchema<S extends Schema = Schema>(doc: Document, schema: S): Promise<boolean> {
   for (const [prop, type] of Object.entries(schema)) {
+    const value = doc[prop]
+
+    if (typeof type === 'string' && type.endsWith('[]')) {
+      if (!Array.isArray(value)) {
+        return false
+      }
+      const expectedType = type.slice(0, -2)
```

Comment:
This can be avoided by using dictionary lookups

Reply:
Done